### PR TITLE
Update hosts.php

### PR DESCRIPTION
**Pull request by LiFiCode**
Add option to show only active host devices

### CODE REVIEW

**Comment by blacksenator** [comment]
Hello @LiFiCode, I appreciate your input. Giving out only the active devices is a good idea. Unfortunately, I don't really like the solution, because output is either a SimpleXMLElement or an array (of SimpleXMLElement). Please find a solution so that the same type is output in both cases.

Hint: https://stackoverflow.com/questions/9643116/deleting-simplexmlelement-node

Regards

Black Senator

**Comment by LiFiCode** [comment]
You are totally right. 
Now outputs SimpleXMLElement instead of array.

I opted for creating a new SimpleXMLElement instead of removing  nodes from the original. The comment Node gets lost on the way - but it is empty anyway.

File: src/hosts.php [hunk 1]
```diff
@@ -356,14 +356,25 @@ public function getMeshList(bool $asJSON = false)
...
      */
-    public function getHostList()
+    public function getHostList(bool $onlyActive = false, string $interfaceType = '')
     {
         $url = $this->getServerAdress() . $this->x_AVM_DE_GetHostListPath();
-
-        return simplexml_load_file($url);
+        $hostList = simplexml_load_file($url);
+        if ($onlyActive) {
+            $hostListxml = '<?xml version="1.0"?><Item>';
```
Comment (blacksenator) [inline src/hosts.php]:
```
        $hostListXML = '<?xml version="1.0" encoding="UTF-8"?><List>';
            foreach ($hostList->xpath('Item[Active="1"]') as $host) {
                if (empty($interfaceType)
                    || (string)$host->InterfaceType == $interfaceType) {
                    $hostListXML .= $host->asXML();
                }
                return new \SimpleXMLElement($hostListXML . '</List>');
            }

        return $hostList;
```

File: src/hosts.php [hunk 2]
```diff
@@ -356,14 +356,25 @@ public function getMeshList(bool $asJSON = false)
      * getHostList
      *
      * returns the list of host devices available from the location given with
-     * x_AVM_DE_GetHostListPath()
+     * x_AVM_DE_GetHostListPath(). You can choose if the output should contain
+     * all known devices (default) or only currently active host devices optionally
+     * filtered by interface type.
      *
+     * @param bool $onlyActive
```
Comment (blacksenator) [inline src/hosts.php]: ` * @param bool $active`

File: src/hosts.php [hunk 3]
```diff
@@ -356,14 +356,25 @@ public function getMeshList(bool $asJSON = false)
...
+     * @param string $interfaceType [802.11|Ethernet|HomePlug]
      * @return simpleXMLElement
      */
-    public function getHostList()
+    public function getHostList(bool $onlyActive = false, string $interfaceType = '')
```
Comment (blacksenator) [inline src/hosts.php]: `bool $active = false,`

File: src/hosts.php [hunk 4]
```diff
@@ -356,14 +356,25 @@ public function getMeshList(bool $asJSON = false)
...
     {
         $url = $this->getServerAdress() . $this->x_AVM_DE_GetHostListPath();
-
-        return simplexml_load_file($url);
+        $hostList = simplexml_load_file($url);
+        if ($onlyActive) {
```
Comment (blacksenator) [inline src/hosts.php]: ìf (acticve) {`

**Comment by LiFiCode** [comment]
Made the changes, thanks for your feedback and of course the project itself...

File: src/hosts.php [hunk 5]
```diff
@@ -360,20 +360,20 @@ public function getMeshList(bool $asJSON = false)
...
      * @return simpleXMLElement
      */
-    public function getHostList(bool $onlyActive = false, string $interfaceType = '')
+    public function getHostList(bool $active = false, string $interfaceType = '')
     {
         $url = $this->getServerAdress() . $this->x_AVM_DE_GetHostListPath();
         $hostList = simplexml_load_file($url);
-        if ($onlyActive) {
+        if ($active) {
             $hostListxml = '<?xml version="1.0"?><Item>';
```
Comment (blacksenator) [inline src/hosts.php]: `$hostListxml = <?xml version="1.0" encoding="UTF-8"?><List>`

File: src/hosts.php [hunk 6]
```diff
@@ -360,20 +360,20 @@ public function getMeshList(bool $asJSON = false)
...
             foreach($hostList->xpath('Item[Active="1"]') as $host)
                 if(empty($interfaceType) || (string) $host->InterfaceType == $interfaceType)
                     $hostListxml .= $host->asXML();
-            return new \SimpleXMLElement($hostListxml.'</Item>');
+            return new \SimpleXMLElement($hostListxml . '</Item>');
```
Comment (blacksenator) [inline src/hosts.php]: `return new \SimpleXMLElement($hostListxml . '</List>');`

**Comment by LiFiCode** [comment]
Looks very much cleaner, of course. But I didn't want to go over the line length limit...

> ... Code MUST follow a "coding style guide" PSR [PSR-1]. Code MUST use 4 spaces for indenting, not tabs. There MUST NOT be a hard limit on line length; the soft limit MUST be 120 characters; lines SHOULD be 80 characters or less.

**Comment by blacksenator** [comment]
Sorry, I don´t know what you try to tell me. I appreciate coding style PSR. The last thing I mentioned is that you used a wrong node. It has to be: 
```PHP
<?xml version="1.0" encoding="UTF-8"?>
<List>
    <Item>
    ...
    </Item>
</List>
```
and not
```PHP
<?xml version="1.0"?>
<Item>
    <Item>
    ...
    </Item>
</Item>

**Comment by LiFiCode** [comment]
i really missed that, i thought you were talking about the split up xml...
gonna fix that now.

**Review by blacksenator — APPROVED** [review]

**Pull request merged**